app/src/main/java/com/infomaniak/mail/ui/main/newMessage/AiViewModel.kt

```diff
@@ -23,29 +23,41 @@ import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.viewModelScope
 import com.infomaniak.mail.data.api.ApiRepository
 import com.infomaniak.mail.di.IoDispatcher
+import com.infomaniak.mail.ui.main.newMessage.AiViewModel.PropositionStatus.*
+import com.infomaniak.mail.utils.ErrorCode.MAX_TOKEN_REACHED
 import com.infomaniak.mail.utils.coroutineContext
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @HiltViewModel
-class AiPromptViewModel @Inject constructor(
+class AiViewModel @Inject constructor(
     application: Application,
     @IoDispatcher private val ioDispatcher: CoroutineDispatcher,
 ) : AndroidViewModel(application) {
 
     private val ioCoroutineContext = viewModelScope.coroutineContext(ioDispatcher)
 
-    val aiProposition = MutableLiveData<String>()
+    val aiProposition = MutableLiveData<Pair<PropositionStatus, String?>?>()
 
     fun generateAiProposition(prompt: String) = viewModelScope.launch(ioCoroutineContext) {
         with(ApiRepository.generateAiProposition(prompt)) {
-            if (isSuccess()) {
-                data?.content.let(aiProposition::postValue)
-            } else {
-                TODO()
-            }
+            aiProposition.postValue(
+                when {
+                    isSuccess() -> data?.content?.let { SUCCESS to it } ?: (MISSING_CONTENT to null)
+                    error?.code == MAX_TOKEN_REACHED -> MAX_TOKEN_EXCEEDED to null
+                    else -> ERROR to null
+                }
+            )
         }
     }
+
+    enum class PropositionStatus {
+        SUCCESS,
+        ERROR,
+        MAX_TOKEN_EXCEEDED,
+        RATE_LIMIT_EXCEEDED,
```

Comment:
RATE_LIMIT_EXCEEDING never seems to be assigned, is it normal ?

Reply:
It's a prevision for the future, this needs to be handled it's just not handled yet

Reply:
Then we should put a TODO in the `when` to avoid forgetting about this before merging in master